support-files/mariadb-socket-convert

```diff
@@ -0,0 +1,155 @@
+#!/bin/bash
+#
+# Used to generate a mariadb.socket file based on the curent mysql/maridb settings
+#
+# This is to assist distro maintainers in migrating to systemd socket activation from
+# a user system with networking settings somewhere in the my.cnf files.
+#
+# Redirect output to user directory like /etc/systemd/system/mariadb.socket.d/migrated-from-my.cnf-settings.conf
+
+
+print_ports()
+{
+if [ -n "${port}" ]; then
+  echo "ListenStream=${bind}${port}"
+fi
+if [ -n "${extraport}" ]; then
+  echo "ListenStream=${bind}${extraport}"
+fi
+}
+
+bindaddress=
+port=
+extraport=
+backlog=
+maxconnections=
+skip_networking=
+socket=
+
+coproc mysqldefaults { /usr/bin/my_print_defaults --mysqld ; }
+while read var; do
+  key=${var%%=*}
+  val=${var#*=}
+  case "${key}" in
+    --bind-address)
+      bindaddress=$val
+      ;;
+    --bind_address)
+      bindaddress=$val
+      ;;
+    --port)
+      port=$val
+      ;;
+    --extra-port)
+      extraport=$val
+      ;;
+    --extra_port)
+      extraport=$val
+      ;;
+    --backlog)
+      backlog=$val
+      ;;
+    --max-connections)
+      maxconnections=$val
+      ;;
+    --max_connections)
+      maxconnections=$val
+      ;;
+    --skip-networking)
+      if [ -z "${val}" ]; then
+        skipnetworking=1
+      else
+        skipnetworking=$val
+      fi
+      ;;
+    --skip_networking)
+      if [ -z "${val}" ]; then
+        skipnetworking=1
+      else
+        skipnetworking=$val
+      fi
+      ;;
+    --socket)
+      socket=$val
+      ;;
+  esac
+done < /dev/fd/${mysqldefaults[0]}
```

Comment:
will a small bit of hacking of mysqld_safe.sh, created mariadb-service-convert that uses mysqld_safe to do the grunt of the processing.

https://github.com/openquery/mariadb-server/blob/e65f3d692182ee5756074e8c9efd39ae14ef99b1/scripts/mariadb-service-convert